kratos/utilities/parallel_levelset_distance_calculator.h

```diff
@@ -708,265 +670,198 @@ class ParallelDistanceCalculator
     ///@name Private Operations
     ///@{
 
-    void Check(ModelPart& rModelPart,
-                            const Variable<double>& rDistanceVar,
-                            const Variable<double>& rAreaVar)
+    void Check(
+        ModelPart& rModelPart,
+        const Variable<double>& rDistanceVar,
+        const Variable<double>& rAreaVar)
     {
         KRATOS_TRY
 
         bool is_distributed = false;
-        if(rModelPart.GetCommunicator().TotalProcesses() > 1)
+        if(rModelPart.GetCommunicator().TotalProcesses() > 1) {
             is_distributed = true;
-
-        const int node_size = rModelPart.Nodes().size();
+        }
 
         //check that variables needed are in the model part
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"distance Variable is not in the model part","");
-
-        if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar)) )
-            KRATOS_THROW_ERROR(std::logic_error,"Area Variable is not in the model part","");
-
-        if(is_distributed == true)
-            if(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX)) )
-                 KRATOS_THROW_ERROR(std::logic_error,"PARTITION_INDEX Variable is not in the model part","")
+        const int node_size = rModelPart.Nodes().size();
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rDistanceVar))) << "Distance variable is not in the model part." << std::endl;
+        KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(rAreaVar))) << "Area variable is not in the model part." << std::endl;
+        if (is_distributed == true) {
+            KRATOS_ERROR_IF(node_size && !(rModelPart.NodesBegin()->SolutionStepsDataHas(PARTITION_INDEX))) << "Area variable is not in the model part." << std::endl;
+        }
```

Comment:
same as above
maybe you could even refactor to a `Check` function?

Reply:
My idea is to refactor this to a `Process` (as the other level set algorithms). I'd do it in there.